File: M338-counting-bits.py
class Solution(object):
    def countBits(self, num):
        """
        :type num: int
        :rtype: List[int]
        """
        bs = [0] * (num + 1)
        for i in range(1, num + 1):
            bs[i] = bs[i&(i-1)] + 1
        return bs

File: Java/M338_CountingBits.java
class Solution {
    public int[] countBits(int num) {
        int[] bcount = new int[num + 1];
        for (int i = 1; i<= num; i++) {
            int l = i & i - 1;
            bcount[i] = bcount[l] + 1;
        }
        return bcount;
    }
}